filters/include/pcl/filters/passthrough.h

```diff
@@ -195,15 +166,45 @@ namespace pcl
       void
       applyFilter (std::vector<int> &indices) override
       {
-        applyFilterIndices (indices);
+        indices.resize (indices_->size());
+        removed_indices_->resize (indices_->size ());
+
+        // Attempt to get the field name's index
+        std::vector<pcl::PCLPointField> fields;
+        int distance_idx = pcl::getFieldIndex<PointT> (filter_field_name_, fields);
+        if (!filter_field_name_.empty() && distance_idx == -1)
+        {
+          PCL_WARN ("[pcl::%s::applyFilter] Unable to find field name in point type.\n", getClassName ().c_str ());
+          indices.clear ();
+          removed_indices_->clear ();
+          return;
+        }
+
+        const auto condition = [&](const PointCloud& cloud, index_t idx) {
+          if (std::isfinite (cloud[idx].x) && std::isfinite (cloud[idx].y) && std::isfinite (cloud[idx].z)) {
+            // If a field name been specified, then filter non-finite entries
+            if (filter_field_name_.empty ())
+              return !negative_;
+
+            // Get the field's value
+            const auto* pt_data = reinterpret_cast<const std::uint8_t*> (&cloud[idx]);
+            float field_value = 0;
+            memcpy (&field_value, pt_data + fields[distance_idx].offset, sizeof (float));
+            if (std::isfinite (field_value) && field_value >= filter_limit_min_ && field_value <= filter_limit_max_) {
+              return true;
+            }
+          }
+          return false;
+        };
+
+        pcl::FunctorFilter<PointT, decltype(condition)> filter{condition, extract_removed_indices_};
+        filter.setInputCloud(input_);
+        filter.setIndices(indices_);
+        filter.setNegative(negative_);
+        filter.applyFilter(indices);
+        removed_indices_ =  std::const_pointer_cast<Indices>(filter.getRemovedIndices());
       }
 
-      /** \brief Filtered results are indexed by an indices array.
-        * \param[out] indices The resultant indices.
-        */
-      void
-      applyFilterIndices (std::vector<int> &indices);
```

Comment:
Keep the stub with a warning and indirection

Reply:
Deprecation in v1.13?